lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -546,84 +574,188 @@ defmodule ExUnit.DocTestTest do
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                code:  "Hello, world" = "Hello world"
+                left:  "Hello, world"
+                right: "Hello world"
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 9}: (test)
+           """
+
+    assert output =~ """
+             4) doctest module ExUnit.DocTestTest.PatternMatching (4) (ExUnit.DocTestTest.PatternMatchingRunner)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                code:  "Hello, " <> _ = "Hello world"
+                left:  "Hello, " <> _
+                right: "Hello world"
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 11}: (test)
+           """
+
+    assert output =~ """
+             5) doctest module ExUnit.DocTestTest.PatternMatching (5) (ExUnit.DocTestTest.PatternMatchingRunner)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                code:  [:a | _] = [:b, :a]
+                left:  [:a | _]
+                right: [:b, :a]
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 13}: (test)
+           """
+
+    assert output =~ """
+             6) doctest module ExUnit.DocTestTest.PatternMatching (6) (ExUnit.DocTestTest.PatternMatchingRunner)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                The following variables were pinned:
+                  atom = :a
+                code:  [^atom | _] = [:b, :a]
+                left:  [^atom | _]
+                right: [:b, :a]
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 16}: (test)
+           """
+
+    assert output =~ """
+             7) doctest module ExUnit.DocTestTest.PatternMatching (7) (ExUnit.DocTestTest.PatternMatchingRunner)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                code:  %{b: _, d: :e} = %{a: :c, d: :e}
+                left:  %{b: _, d: :e}
+                right: %{a: :c, d: :e}
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 18}: (test)
+           """
+
+    assert output =~ """
+             8) doctest module ExUnit.DocTestTest.PatternMatching (8) (ExUnit.DocTestTest.PatternMatchingRunner)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                match (=) failed
+                code:  %{year: 2001, day: 1} = ~D"2000-01-01"
+                left:  %{year: 2001, day: 1}
+                right: ~D[2000-01-01]
+                stacktrace:
+                  (for doctest at) test/ex_unit/doc_test_test.exs:#{starting_line + 20}: (test)
```

Comment:
We don't have any tests ensuring that the two doctests that should pass did indeed pass. Should we add something here making sure we have 2 successful tests?